### ISSUE TITLE
[GR-54068] Implement Relaxed SIMD Proposal for WebAssembly

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9759]
### TL;DR

We plan to implement [the WebAssembly Relaxed SIMD proposal](https://github.com/WebAssembly/relaxed-simd) in GraalWasm.

### Details

The Relaxed SIMD proposal builds on top of the existing [fixed-width SIMD proposal](https://github.com/WebAssembly/simd) by providing new instructions that work on vector values. These new arithmetic instructions are non-deterministic - given the same inputs, two calls to the same instruction can return different results. The non-determinism of these instructions makes them more portable, as different machine architectures might have corresponding primitives that return different results. In such cases, if a deterministic instruction was used, only some architectures would be able to use the host primitive directly, while others would have to calculate the result in a suboptimal manner. This non-specificity thus makes it easier for WebAssembly runtimes to deliver consistent performance across various architectures.

More details at https://github.com/WebAssembly/relaxed-simd/blob/main/proposals/relaxed-simd/Overview.md
